Src/Support/GoogleApis.Tests_dotnetcore/System.Net/AsyncStream.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net
+{
+    class AsyncStream
+    {
+        public AsyncStream(Stream stream)
+        {
+            _stream = stream;
+        }
+
+        private readonly Stream _stream;
+
+        private byte[] _buffer = new byte[1024];
+        private int _bufferCount = 0;
+        private int _bufferOffset = 0;
+
+        // Read bytes until the predicate returns false.
+        // If the end of stream is reached, the bytes read so far are returned.
+        public async Task<IEnumerable<byte>> ReadWhile(Func<List<byte>, bool> predicate, CancellationToken ct = default(CancellationToken))
```

Comment:
I'll see this later, but this is done so the caller can determine if a _separator_ has been found and stop reading from the stream. Can it just be passing the byte in question instead of the growing buffer?